### PR TITLE
Add a short guide on code completion for Unity in VS code

### PR DESCRIPTION
**Pull request by Juksefantomet**
As the original guide does not implement this specific case of allowing visual code to hook into code completion from unity assets i added a short descriptive guide that i just completed and can confirm is working.

I did this on a system that was formatted and reinstalled yesterday without any prior installs. so a clean system does in fact require this to properly hook into code completion.

### CODE REVIEW

**Comment by sailro** [comment]
Hi @Juksefantomet and thank you for this contribution. 

My initial feedback on this:

- You don't need the full VS to install a .NET targeting pack. Those are available as standalone downloads. So perhaps it's more convenient for our users to use the direct download URL.
- I'm not sure 4.5.x is the best to recommend. First Unity 2017.x releases started to reference .NET 4.6 assemblies then switched to 4.7.x after.
- This is only needed when targeting the new Unity run-time. It's not useful for the previous .NET 2.x run-time. (But indeed, the old runtime will be deprecated with 2019.x).
- There is a already a question regarding Intellisense in "Common questions" just below your paragraph. 

I'm more used to the full VS experience, but I will check tomorrow how everything works with VS Code.
I just want to be sure that we are not missing something here.

Regards
Sebastien Lebreton [MSFT]
Visual Studio Tools for Unity

**Comment by Juksefantomet** [comment]
Greetings!

1. I installed Visual studio code
2. I installed Unity -> VS was then installed automatically on a default setup
3. Running latest Unity version, latest visual studio code version and editing a script out of the box this was the guide and solution that immediately solved my issue.

I was mainly doing this because a friend of mine was not figuring out how to do it, and as a developer i am more familiar with certain troubleshooting and guides on how to solve issues in relation to some problems that are presented.

As my log in visual studio code directly reference 4.5 packages and outputted errors in relation to that targeted version that was my first option and it worked. 😄 

I appreciate the feedback and absolutely, if i'm going overboard here no problem at all. Give me pointers to properly adapt it, and/or submit some feedback on changes and i will gladly adjust them accordingly.

But do keep in a mind, a fresh clean format + install of windows 10 pro, 64-bit, running fresh install of vs code and unity latest version. as of 2 days ago. Doing everything by default settings resulted in the presented guide without any out of the ordinary steps.

I am a fan of properly presented guides that are accurate and on point, so by all means i was hoping for feedback on this.

**Comment by Juksefantomet** [comment]
@sailro 

Sorry forgot to address the final part.

The Intellisense does work from start. but the 4.5.x targeting packs allows VSCode to properly hook into the unity core library for code suggestions/codecompletion. 

That was the main problem and that's what was not showing up without applying the targeting packs. i was researching this for a well spent 2+ hours before i figured out how to solve it. so it is by no means a "clear solution" to this issue.

**Comment by sailro** [comment]
Ok understood. 

You had an omnisharp error in the log right ? 

See our friend @DustinCampbell solved the exact same issue here:
https://github.com/OmniSharp/omnisharp-vscode/issues/2010

So what do you think of updating the PR reusing what is given in the issue resolution above (direct download of the targeting pack on Windows (Instead of using VS) or .NET SDK on MacOS+Linux).

@DustinCampbell does this make sense or are we missing something ?

Thanks!
Seb

**Comment by Juksefantomet** [comment]
@sailro 

I can confirm that the output error is one of the errors that lead me to the solution. The other one was related to CLI but that was irrelevant for me at the time.

As this didn't include any relations to code completion i stumbled across the log error 😄 i never actually found any "results" when searching that would confirm the targeting pack would fix my issue. But it clearly does have a direct relation so i will update accordingly.

Just to clarify:

1. I assume we want a link which reflects a "your 4.X.X targeting pack" target download
2. Remove references to "VS" usage in acquisition of the targeting packs and/or SDK/.net packs.

Correct?
-

If there is a preferred landing site for acquiring all of the above mentioned packs, would you be so kind as to clarify that link which i can provide in the PR?

**Comment by sailro** [comment]
Yes this is correct.

Regarding the URL, I think the best is to use the download link provided in the omnisharp issue:
For the targeting pack on Windows: https://www.microsoft.com/en-us/download/details.aspx?id=48136
For the .NET SDK on MacOS: https://dotnet.microsoft.com/learn/dotnet/hello-world-tutorial#macos

Thanks!
Seb

**Comment by Juksefantomet** [comment]
Hello!

Sorry about my delayed reply but my graphics card of all things died. I will update this according to agreed changes as soon as i have time to do so and resubmit.

Thanks for the feedback and happy to be of assistance!

**Comment by Juksefantomet** [comment]
@sailro 

There we go! I've rephrased and adapted the guide according to feedback and provided direct links!

Good to go!

**Comment by sailro** [comment]
@Juksefantomet 

I spotted a little typo (-acquire-). Apart from that LGTM.

cc @gregvanl for approval

Thanks!

File: docs/other/unity.md [hunk 1]
```diff
@@ -66,6 +66,23 @@ The community is continually developing more and more valuable extensions for Un
 
 The extensions shown above are dynamically queried. Click on an extension tile above to read the description and reviews to decide which extension is best for you. See more in the [Marketplace](https://marketplace.visualstudio.com/vscode).
 
+## Adding code completion to visual code (For recent version of Unity)
+
+If you are installing VSCode for the first time you might be missing targeting packs which unlocks Unitys code-completion in VSCode.
+
+Here are the provided links to aquire the targeting packs:
```
[the remark this answers is not in the document]
Comment (gregvanl) [reply docs/other/unity.md]: I can fix this typo

**Comment by gregvanl** [comment]
Thanks! I did a little editing with https://github.com/Microsoft/vscode-docs/commit/879f6fcba5d80e9e2d7ecbcc00a4443da005755f

**Pull request merged**